File: Selenium/Activities/Activity18.java
package Activities;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

public class Activity18 {
    public static void main(String[] args) throws InterruptedException {
        WebDriverManager.firefoxdriver().setup();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriver newDriver = new FirefoxDriver();
        newDriver.get("https://training-support.net/selenium/selects");
        System.out.println("Title of page: " + newDriver.getTitle());
        WebElement dropdown = newDriver.findElement(By.id("multi-select"));
        Select multiSelect = new Select(dropdown);
        multiSelect.selectByVisibleText("Javascript");
        multiSelect.selectByIndex(3);
        multiSelect.selectByIndex(4);
        multiSelect.selectByIndex(5);
        multiSelect.selectByValue("node");
        multiSelect.deselectByIndex(4);
        List<WebElement> allOptions = multiSelect.getAllSelectedOptions();
        for(WebElement item : allOptions)
            System.out.println(item.getText() + " ");
        Thread.sleep(300);
        newDriver.quit();
    }
}

File: Selenium/Activities/Activity18.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    driver.get("https://training-support.net/selenium/selects")
    print("Title is: ", driver.title)
    select = Select(driver.find_element(By.ID, "multi-select"))
    select.select_by_visible_text("Javascript")
    select.select_by_index(3)
    select.select_by_index(4)
    select.select_by_index(5)
    select.select_by_value("node")
    select.deselect_by_index(4)
    for item in select.all_selected_options:
        print(item.text)
    driver.quit()